frame/staking/src/lib.rs

```diff
@@ -489,7 +491,7 @@ where
 				let normal_ring = *active_ring - *active_deposit_ring;
 				if normal_ring < *slash_ring {
 					let mut slash_deposit_ring = *slash_ring - (*active_ring - *active_deposit_ring);
-					*active_deposit_ring -= slash_deposit_ring;
+					*active_deposit_ring = active_deposit_ring.saturating_sub(slash_deposit_ring);
```

Comment:
I need to check this.

Reply:
This related to the testcase `reward_validator_slashing_validator_doesnt_overflow`, we can add additional test for this, but this pr is quite large currently